nomad/client_alloc_endpoint.go

```diff
@@ -41,16 +42,29 @@ func (a *ClientAllocations) GarbageCollectAll(args *structs.NodeSpecificRequest,
 		return errors.New("missing NodeID")
 	}
 
+	// Make sure Node is valid and new enough to support RPC
+	snap, err := a.srv.State().Snapshot()
+	if err != nil {
+		return err
+	}
+
+	node, err := snap.NodeByID(nil, args.NodeID)
```

Comment:
I wonder if we could benefit from a helper to keep the code a little more DRY:
```
func getNodeForRpc(snap *state.StateSnapshot, nodeID string) (*structs.Node, error) {
  node, err := snap.NodeByID(nil, nodeID)
  if err != nil {
    return nil, err
  }
  
  if node == nil {
    return nil, fmt.Errorf("Unknown node %q", nodeID)
  }

  if err := nodeSupportsRpc(node); err != nil {
      return nil, err
  }

  return node, nil
}
```

Reply:
Good call. Worked out perfectly.